wpcom-legacy-redirector.php

```diff
@@ -104,13 +105,58 @@ static function insert_legacy_redirect( $from_url, $redirect_to ) {
 			return new WP_Error( 'invalid-redirect-url', 'Invalid redirect_to param; should be a post_id or a URL' );
 		}
 
-		wp_insert_post( $args );
+		switch ( $mode ) {
+			case 'update':
+				$args[ 'ID' ] = self::get_redirect_post_id( $from_url, false );
```

Comment:
the `self::redirect_post_id()` may return `0` which would be ignoring the update request. So perhaps we should check the return value of that call before proceeding with the update procedure, since it would end up in adding new redirect, instead of updating existing one, which may not be the action which has been intended by the user.